site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3331,6 +3331,21 @@ public function getRegistrationSections() {
         return $this->course_db->rows();
     }
 
+    /**
+     * Gets the default registration section for a given course and term from the courses table
+     */
+    public function getDefaultRegistrationSection(string $term, string $course): string|null {
+        $this->submitty_db->query("SELECT default_section_id FROM courses where term=? and course=?", [$term, $course]);
+        return $this->submitty_db->row()['default_section_id'];
```

Comment:
Whether or not there is a check in the controller which provides user-facing error messages depends on how thorough you want to be.  Having a check here would simply be for robustness purposes, since this function shouldn't care about how errors are actually handled.  We just don't want this to crash unexpectedly if the database gets in a funny state where no rows are returned.

Reply:
After discussion with Barb, this isn't a priority issue, while it may 'break' something if the database gets in a bad state, it is a lower priority thing due to most users using the UI and not manually editing the database.

Reply:
Couldn't you easily avoid an error in this function by adding `?? null` to the return line so it returns null if the array key doesn't exist?